mobilizon_reshare/publishers/platforms/mastodon.py

```diff
@@ -0,0 +1,129 @@
+import pkg_resources
+import requests
+from requests import Response
+
+from mobilizon_reshare.event.event import MobilizonEvent
+from mobilizon_reshare.publishers.abstract import (
+    AbstractPlatform,
+    AbstractEventFormatter,
+)
+from mobilizon_reshare.publishers.exceptions import (
+    InvalidBot,
+    InvalidCredentials,
+    InvalidEvent,
+    InvalidResponse,
+    PublisherError,
+    ServerError,
+    ClientError,
+)
+from mobilizon_reshare.publishers.platforms.utils import uri_join, fqdn_to_uri
+
+
+class MastodonFormatter(AbstractEventFormatter):
+
+    _conf = ("publisher", "mastodon")
+    default_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon.tmpl.j2"
+    )
+
+    default_recap_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon_recap.tmpl.j2"
+    )
+
+    default_recap_header_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon_recap_header.tmpl.j2"
+    )
+
+    def validate_event(self, event: MobilizonEvent) -> None:
+        text = event.description
+        if not (text and text.strip()):
+            self._log_error("No text was found", raise_error=InvalidEvent)
+
+    def validate_message(self, message) -> None:
+        if len(message.encode("utf-8")) >= 500:
+            raise PublisherError("Message is too long")
+
+
+class MastodonPlatform(AbstractPlatform):
+    """
+    Mastodon publisher class.
+    """
+
+    _conf = ("publisher", "mastodon")
+    api_uri = "api/v1/"
+
+    def _send(self, message: str) -> Response:
+        """
+        Send messages
+        """
+        return requests.post(
+            url=uri_join(fqdn_to_uri(self.conf.instance), self.api_uri) + "statuses",
+            headers={"Authorization": f"Bearer {self.conf.token}"},
+            data={
+                "status": message,
+                "visibility": "public",
+            },
+        )
+
+    def validate_credentials(self):
+        conf = self.conf
```

Comment:
I removed this from the other publishers. It's unnecessary because the config is validated in the beginning. Check how they look like now